sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_authentication_async.py

```diff
@@ -76,8 +84,8 @@ async def on_challenge(
         :returns: a bool indicating whether the policy should send the request
         :rtype: bool
         """
-
-        challenge = response.http_response.headers.get("WWW-Authenticate")
+        # Casting, as the code seems to be certain that on_challenge this header will be present
+        challenge: str = cast(str, response.http_response.headers.get("WWW-Authenticate"))
```

Comment:
nit: not sure if we need the inline type hint if we use cast

Reply:
I don't think it hurts, so I'm keeping it to no re-change the PR :p